wdl-analysis/tests/analysis/name-reference-cycle/source.diagnostics

```diff
@@ -0,0 +1,9 @@
+error: name reference cycle detected
+  ┌─ tests/analysis/name-reference-cycle/source.wdl:6:9
+  │
+6 │     Int a = b
+  │         ^ this name participates in the cycle
+7 │     Int b = c
```

Comment:
Should this be flagged as participating in the cycle?

Reply:
The DFS stops at the first place it encounters a cyclic reference, so we only report on two locations: the starting point and the node that we've encountered which was previously visited.

If we want, we could walk _a_ path between the two nodes and report on each node we encounter.